### ISSUE TITLE
1. Dedicated spawn points for each unit type (air, ground, naval).  AND Waypoints 

### ISSUE THREAD

**Issue by Meatsim009** [Anuken/Mindustry-Suggestions#1272]
**Describe the content or mechanics you are proposing.**

1. Establishment of unit type dedicated spawn points for air, ground and naval units
2. Waypoints for pathing

**Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.**

1.  Dedicated spawn points.

In my case, I have created a survival map which only has 1 1x1 red spawn point area.  I've purposefully done this to keep control of unit / wave scaling.  However, both naval and ground mechs have to spawn at the same place, which means having water underlying the spawn point.  An increased number of naval units eventually start pushing ground units into deep water and they perish.

By being able to set a dedicated naval unit spawn point, the user can make sure all naval units spawn on water and won't have the tendency to push ground units into deep water and their deaths.  Ground units can have their own dedicated spawn point which avoids naval being spawned on land.

The opportunities this could lead to in having dedicated "attack fronts" - having different unit types pose different threats from different points on the map.

The red 'S' logo used in the editor for spawn points could be changed to [red]S[green]A for air unit spawn, [red]S[yellow]G for ground unit spawn, and [red]S[blue]N for naval unit spawn.

2.  Waypoints

In some cases, where there is a maze for enemy naval units to follow to get to the nearest enemy core to destroy it, the naval pathing can end up being stuck against a wall, even though there is an indirect (longer) path for the naval units to reach the player teams core.

Having the option to set way points for enemy units to follow in the editor can initially help units know which area they must travel to first as a priority, before they return to 'general' pathing.

As an example, on my map https://steamcommunity.com/sharedfiles/filedetails/?id=2240728575 (which was purposefully created as a sandbox / survival map to show off your beta developments), the ground mechs and the naval mechs have a windy path to take to get to the players cores.  The ground mechs navigate this easily, but the naval mechs get stuck against a wall and refuse to travel the length of the river.

Being able to set waypoints in the editor may alleviate such pathing issues.

It could benefit the player team getting around this issue also without having to place multiple command centres or using logic to overcome pathing issues.

thank you for your consideration.  Proud to support your game on Steam.

Meatsim009
maker of...
FortKnoxRPG
onthebleedingedge
footy 3k(WIP)

**Before making this issue, place an `X` in the boxes below to confirm that you have acknowledged them.** *Failure to do so may result in your request being closed automatically.*

1. - [x] I have done a quick search in the list of suggestions to make sure this has not been suggested yet.
2. - [x] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
3. - [x] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
4. - [x] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by itcannotbe** [comment]
[X to doubt] I have done a quick search in the list of suggestions to make sure this has not been suggested yet.

**Comment by Meatsim009** [comment]
> [X to doubt] I have done a quick search in the list of suggestions to make sure this has not been suggested yet.
> 
> #325

i had a quick search in the suggestions but i did not see anything, unless it is the one I did earlier today that I didn't think went through properly.  I am new to github

**Comment by Meatsim009** [comment]
I've gone through over 10 pages in the the suggestions area and haven't seen my idea in the title.  If it has already been suggested, please point me in that direction.  First day at github

**Comment by SupremeTurtle** [comment]
It was 4 months ago, so it can be easily forgiven.

**Comment by scarf-generic** [comment]
> I've gone through over 10 pages in the the suggestions area and haven't seen my idea in the title. If it has already been suggested, please point me in that direction. First day at github

In the future, its useful to search for common words you used. For example "waves" "pathfinding" "spawn" "waypoint" etc. Helps to narrow down your search a bit so you only get relevant posts to look through

**Comment by itcannotbe** [comment]
Right....... A search function doesn't exist on github..................

**Comment by RedstonerKBC** [comment]
I made a suggestion similar to this many months ago too. Basically the idea was to have custom waves for each spawnpoint. The mapmaker can choose to make say multiple spawn rules then distribute them as they wish among spawnpoints.

**Comment by Meatsim009** [comment]
> > I've gone through over 10 pages in the the suggestions area and haven't seen my idea in the title. If it has already been suggested, please point me in that direction. First day at github
> 
> In the future, its useful to search for common words you used. For example "waves" "pathfinding" "spawn" "waypoint" etc. Helps to narrow down your search a bit so you only get relevant posts to look through

Thank you.

> Right....... A search function doesn't exist on github..................

as I said, first day on github.  I haven't purposefully tried to evade the conditions set in the template.  That being said, I finally found and used the search function and found yours.  As such, I can understand your frustration at this noob 👍 

> I made a suggestion similar to this many months ago too. Basically the idea was to have custom waves for each spawnpoint. The mapmaker can choose to make say multiple spawn rules then distribute them as they wish among spawnpoints.

I managed to find and use the search function and can see that similar posts were made

https://github.com/Anuken/Mindustry-Suggestions/issues/898 - @Humanoid-X
https://github.com/Anuken/Mindustry-Suggestions/issues/325 - @itcannotbe 

HOWEVER, I could not find a post dedicated to waypoints in the way I have suggested in regards to enemy units.  the closest I found was this...

https://github.com/Anuken/Mindustry-Suggestions/issues/44

As such, I would like to keep my request open as is.  

But, credit where it is due.  To all of you who have made these requests in whole or in part before, I apologise for not being able to properly use github and search properly and - in doing so - to seemingly render your suggestions as unimportant.  I am glad that others have had these ideas before, and I trust that we may see them implemented in the game down the track, with credit given to you.
